todo

```diff
@@ -69,10 +69,10 @@ function push_to_stack {
 }
 
 function find_when {
-    if [[ $(uname -s) == "Darwin" ]]; then
-        WHEN="$(date -j -f %s "$1")"
-    else
+    if date --version 2> /dev/null | grep -q GNU; then
         WHEN="$(date --date "@$1")"
+    else
+        WHEN="$(date -j -f %s "$1")"
     fi
 }
```

Comment:
I don't know what the absolute path is on a Linux box, maybe it is also `/bin/date` but I would still think you'd want to know the version of the binary for sure rather than assume based on a path.

Reply:
it's `/bin/date` . even if it's really `/usr/bin/date` it's still `/bin/date` since most distros are now 'merged-usr'. And frankly, if you installed something else into `/bin/date` you get to keep the pieces :)

i hear what you're saying though. i'll 🤔 it over